wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/ItemDocumentImpl.java

```diff
@@ -142,7 +142,11 @@ protected ItemDocumentImpl(
 	@JsonIgnore
 	@Override
 	public ItemIdValue getEntityId() {
-		return new ItemIdValueImpl(this.entityId, this.siteIri);
+		if (!EntityIdValue.SITE_LOCAL.equals(siteIri)) {
+			return new ItemIdValueImpl(entityId, siteIri);
+		} else {
+			return ItemIdValue.NULL;
+		}
```

Comment:
This change was made for consistency with `FormDocumentImpl`. `FormDocumentImpl` needed the change, because `FormIdValueImpl` constructor throws when provided with null form ID. I will modify regexes in `FormIdValueImpl` and `SenseIdValueImpl` to accept null IDs and then revert changes in `getEntityId()`.

Reply:
Ah I see. Maybe that's worth doing in a separate PR so we can assess the change independently? I suspect the reason why this was introduced in FormDocumentImpl is because of the compound structure of the ids there (but I haven't looked at the issue closely).

Reply:
This has to be done now. Otherwise this code in `LexemeUpdateBuilder` would throw spurious exceptions:

```java
	public LexemeUpdateBuilder addForm(FormDocument form) {
		Objects.requireNonNull(form, "Form cannot be null.");
		if (form.getEntityId().isValid())
			form = form.withEntityId(FormIdValue.NULL);
		addedForms.add(form);
		return this;
	}
```

Reply:
I have reverted this change. In order to make the code work, I have modified regexes in `FormIdValueImpl` and `SenseIdValueImpl` to accept placeholder IDs.